Add 'BulkThumbnailCreator' project

BugMixer95 opened the pull request:
This PR contains new application called **Bulk Thumbnail Creator**.
The main purpose of it is to allow user to resize/rename a huge amount of images.
The application also write log entries to Windows Event Viewer and to local file.

BugMixer95 committed:
initial: project creation

BugMixer95 committed:
adding app.config and DirectoryHandler class

BugMixer95 committed:
adding core logic of image processing

BugMixer95 committed:
adding dialogue with user

BugMixer95 committed:
fixing user's inability to choose any action after cancellation

BugMixer95 committed:
small UI changes

BugMixer95 committed:
adding EventLogger class

Now main actions (creating project folders, image resizing and renaming) are logged into Windows Event Viewer

BugMixer95 committed:
adding BTCLogger class

Now main actions (creating project folders, image resizing and renaming) are logged into a separate file. These logs are more detailed than logs for Event Viewer

BugMixer95 committed:
changing project structure

Moving DirectoryHandler and ImageHandler classes to the Modules folder

BugMixer95 committed:
fixing bug when Source Files directory is empty

BugMixer95 committed:
adding DisplayProgress class

DisplayProgress class shows user how much files are processed, total amount of files in Source Files directory and readiness percentage

BugMixer95 committed:
make aborting operation to show actual UserName

BugMixer95 committed:
Merge branch 'dev' into _dev/bulk_thumbnail_creator_dzianis

Audyman commented on Shastak/BulkThumbnailCreator/Extras/BTCLogger.cs:
```diff
@@ -0,0 +1,39 @@
+
+            using (StreamWriter sw = new StreamWriter(LogsFilename, true))
+            {
+                string now = DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss.fffff");
```
Better to move date format to config or to separate class for string constants

BugMixer95 replied:
fixed by adding Constants.cs class

Audyman commented on Shastak/BulkThumbnailCreator/Extras/BTCLogger.cs:
```diff
@@ -0,0 +1,39 @@
+
+        private static void CreateLogsFileForToday()
+        {
+            string today = DateTime.Now.ToString("yyyy-MM-dd");
```
see above

BugMixer95 replied:
fixed by adding Constants.cs class

Audyman commented on Shastak/BulkThumbnailCreator/Extras/BTCLogger.cs:
```diff
@@ -0,0 +1,39 @@
+            {
+                using (StreamWriter sw = new StreamWriter(filename, true))
+                {
+                    string now = DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss.fffff");
```
see above

BugMixer95 replied:
fixed by adding Constants.cs class

Audyman commented on Shastak/BulkThumbnailCreator/Modules/ImageHandler.cs:
```diff
@@ -0,0 +1,229 @@
+                // if image is successfully processed it goes to the processed files directory
+                try
+                {
+                    BTCLogger.WriteLocalLogs($"Starting resizing file {Files[i].Name}");
```
naming: BtcLogger

BugMixer95 replied:
fixed

Audyman commented on Shastak/BulkThumbnailCreator/Modules/ImageHandler.cs:
```diff
@@ -0,0 +1,229 @@
+            {
+                TextColorizer.WriteTextInColor(
+                    "Your source folder doesn't contain any files! \n" +
+                    "Please add some files in the source folder and press any key to continue. \n",
```
Better to move message texts to separate class e.g. Constants

BugMixer95 replied:
fixed by adding Constants.cs class

Audyman commented on Shastak/BulkThumbnailCreator/Program.cs:
```diff
@@ -0,0 +1,130 @@
+
+            Console.Write("Welcome to the ");
+            TextColorizer.WriteTextInColor("BulkThumbnailCreator", ConsoleColor.Green, false);
+            Console.WriteLine("! \nThis program allows you to resize or rename your images.");
```
see above

BugMixer95 replied:
fixed by adding Constants.cs class

Audyman reviewed (approved):
some minor comments, but code base is very good

BugMixer95 committed:
PR comments fix: fixing naming

BTCLogger -> BtcLogger

BugMixer95 committed:
PR comments fix: adding Constants class

Adding Constants class, which acts like a dictionary for text messages and string formats

BugMixer95 committed:
fixing case when rename operation cannot be correctly cancelled

Fixed bug - when rename operation is being cancelled it doesn't actually cancels because of no exit from the loop
